pandas/core/arrays/base.py

```diff
@@ -653,15 +661,59 @@ def copy(self, deep=False):
         raise AbstractMethodError(self)
 
     # ------------------------------------------------------------------------
-    # Block-related methods
+    # Printing
     # ------------------------------------------------------------------------
+    def __unicode__(self):
+        from pandas.io.formats.printing import format_object_summary
+
+        template = (
+            u'<{class_name}>\n'
+            u'{data}\n'
+            u'Length: {length}, dtype: {dtype}'
+        )
+        # the short repr has no trailing newline, while the truncated
+        # repr does. So we include a newline in our template, and strip
+        # any trailing newlines from format_object_summary
+        data = format_object_summary(self, self._formatter(), name=False,
+                                     trailing_comma=False).rstrip()
+        name = self.__class__.__name__
+        return template.format(class_name=name, data=data,
+                               length=len(self),
+                               dtype=self.dtype)
+
+    def _formatter(self, formatter=None):
+        # type: (Optional[ExtensionArrayFormatter]) -> Callable[[Any], str]
+        """Formatting function for scalar values.
+
+        This is used in the default '__repr__'. The formatting function
+        receives instances of your scalar type.
+
+        Parameters
+        ----------
+        formatter: GenericArrayFormatter, optional
+            The formatter this array is being rendered with. The formatter
+            may have a `.formatter` method already defined. By default, this
+            will be used if a `formatter` is passed, otherwise the formatter
+            is ``str``.
```

Comment:
I didn't yet look in detail to the further changes in this PR, but from an EA author view, I find the above not very clear.

In what cases would a `formatter` be passed? As an external EA authord, do I need to care or can I ignore this?

Reply:
Will try to clarify. I believe formatter is passed when we’re boxed in a DataFrame, Series, or Index.

But for a top-level repr(array) no formatter is passed.